src/main/java/com/microsoft/azure/vmagent/remote/AzureVMAgentSSHLauncher.java

```diff
@@ -133,18 +136,31 @@ public void launch(final SlaveComputer agentComputer, final TaskListener listene
             String initScript = agent.getInitScript();
 
             // Executing script only if script is not executed even once
+            String command;
+            if (isUnix) {
+                command = "test -e ~/.azure-agent-init";
+            } else {
+                command = "dir C:\\azure-agent-init";
+            }
             if (StringUtils.isNotBlank(initScript)
-                    && executeRemoteCommand(session, "test -e ~/.azure-agent-init", logger) != 0) {
+                    && executeRemoteCommand(session, command, logger) != 0) {
                 LOGGER.info("AzureVMAgentSSHLauncher: launch: Init script is not null, preparing to execute script remotely");
-                copyFileToRemote(session, new ByteArrayInputStream(initScript.getBytes("UTF-8")), REMOTE_INIT_FILE_NAME);
-
+                if (isUnix) {
+                    copyFileToRemote(session, new ByteArrayInputStream(initScript.getBytes("UTF-8")), REMOTE_INIT_FILE_NAME);
+                } else {
+                    copyFileToRemote(session, new ByteArrayInputStream(initScript.getBytes("UTF-8")), REMOTE_INIT_FILE_NAME_WINDOWS);
+                }
                 // Execute initialization script
                 // Make sure to change file permission for execute if needed. TODO: need to test
 
                 // Grab the username/pass
                 StandardUsernamePasswordCredentials creds = AzureUtil.getCredentials(agent.getVMCredentialsId());
 
-                String command = "sh " + REMOTE_INIT_FILE_NAME;
+                if (isUnix) {
+                    command = "sh " + REMOTE_INIT_FILE_NAME;
+                } else {
+                    command = "powershell " + REMOTE_INIT_FILE_NAME_WINDOWS;
+                }
                 int exitStatus = executeRemoteCommand(session, command, logger, agent.getExecuteInitScriptAsRoot(), creds.getPassword().getPlainText());
```

Comment:
There is a known issue that powershell.exe will always return 0 if it's running a .ps1 file. So checking exit code won't work here.